src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2938,13 +2974,13 @@ private bool HasUnboxingConversion(TypeSymbol source, TypeSymbol destination, re
                 !destination.IsNullableType() &&
                 HasBoxingConversion(destination, source, ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.ValueTypeUnboxing;
```

Comment:
This one is a bit weird.  If destination is a type parameter constrained on struct, and constrained on that interface type, then we'll report this as a ValueTypeUnboxing.  However, if we moved the code below higher:

```c#
if (HasUnboxingTypeParameterConversion(source, destination, ref useSiteDiagnostics))
```

Then we woudl report it as a TypeParameter unboxing.  in this case, it feels weird that we would return ValueTypeUnboxing here. 

Thoughts?